test/server/http/admin_test.cc

```diff
@@ -650,6 +655,103 @@ TEST_P(AdminInstanceTest, TracingStatsDisabled) {
   }
 }
 
+TEST_P(AdminInstanceTest, ClustersJson) {
+  Upstream::ClusterManager::ClusterInfoMap cluster_map;
+  ON_CALL(server_.cluster_manager_, clusters()).WillByDefault(ReturnPointee(&cluster_map));
+
+  NiceMock<Upstream::MockCluster> cluster;
+  cluster_map.emplace(cluster.info_->name_, cluster);
+
+  NiceMock<Upstream::Outlier::MockDetector> outlier_detector;
+  ON_CALL(Const(cluster), outlierDetector()).WillByDefault(Return(&outlier_detector));
+  ON_CALL(outlier_detector, successRateEjectionThreshold()).WillByDefault(Return(6.0));
+
+  ON_CALL(*cluster.info_, addedViaApi()).WillByDefault(Return(true));
+
+  Upstream::MockHostSet* host_set = cluster.priority_set_.getMockHostSet(0);
+  auto host = std::make_shared<NiceMock<Upstream::MockHost>>();
+
+  envoy::api::v2::core::Locality locality;
+  locality.set_region("test_region");
+  locality.set_zone("test_zone");
+  locality.set_sub_zone("test_sub_zone");
+  ON_CALL(*host, locality()).WillByDefault(ReturnRef(locality));
+
+  host_set->hosts_.emplace_back(host);
+  Network::Address::InstanceConstSharedPtr address =
+      Network::Utility::resolveUrl("tcp://1.2.3.4:80");
+  ON_CALL(*host, address()).WillByDefault(Return(address));
+
+  Stats::IsolatedStoreImpl store;
+  store.counter("test_counter").add(10);
+  store.gauge("test_gauge").set(11);
+  ON_CALL(*host, gauges()).WillByDefault(Invoke([&store]() { return store.gauges(); }));
+  ON_CALL(*host, counters()).WillByDefault(Invoke([&store]() { return store.counters(); }));
+
+  ON_CALL(*host, healthy()).WillByDefault(Return(true));
+
+  ON_CALL(host->outlier_detector_, successRate()).WillByDefault(Return(43.2));
+
+  Buffer::OwnedImpl response;
+  Http::HeaderMapImpl header_map;
+  EXPECT_EQ(Http::Code::OK, getCallback("/clusters?format=json", header_map, response));
+  std::string output_json = TestUtility::bufferToString(response);
+  envoy::admin::v2alpha::Clusters output_proto;
+  MessageUtil::loadFromJson(output_json, output_proto);
+
+  const std::string expected_json = R"EOF({
+ "cluster_statuses": [
+  {
+   "name": "fake_cluster",
+   "success_rate_ejection_threshold": {
+    "value": 6
+   },
+   "added_via_api": true,
+   "host_statuses": [
+    {
+     "address": {
+      "socket_address": {
+       "protocol": "TCP",
+       "address": "1.2.3.4",
+       "port_value": 80
+      }
+     },
+     "stats": {
+      "test_counter": {
+       "value": "10",
+       "type": "COUNTER"
+      },
+      "test_gauge": {
+       "value": "11",
+       "type": "GAUGE"
+      },
+     },
+     "health_status": {
+      "eds_health_status": "HEALTHY"
+     },
+     "success_rate": {
+      "value": 43.2
+     }
+    }
+   ]
+  }
+ ]
+}
+)EOF";
+
+  envoy::admin::v2alpha::Clusters expected_proto;
+  MessageUtil::loadFromJson(expected_json, expected_proto);
+
+  // Ensure the protos created from each JSON are equivalent.
+  EXPECT_TRUE(Protobuf::util::MessageDifferencer::Equivalent(output_proto, expected_proto));
```

Comment:
Can you move this to https://github.com/envoyproxy/envoy/blob/20c0454545761c254180f77948304083406757be/test/test_common/utility.h#L156 and then you can take advantage of https://github.com/envoyproxy/envoy/blob/20c0454545761c254180f77948304083406757be/test/test_common/utility.h#L381? This would be a really nice cleanup.